cmds/ocm/commands/ocmcmds/components/hash/cmd.go

```diff
@@ -180,17 +181,26 @@ func (h *action) Close() error {
 
 func (h *action) Out() error {
 	if len(h.norms) > 1 {
-		dir := h.mode.outfile
-		dir = strings.TrimSuffix(dir, ".ncd")
-		err := h.ctx.FileSystem().Mkdir(dir, 0o755)
-		if err != nil {
-			return fmt.Errorf("cannot create output dir %s", dir)
-		}
-		for k, n := range h.norms {
-			p := filepath.Join(dir, k.String())
-			err := h.write(p+".ncd", n)
+		if h.mode.outfile == "" || h.mode.outfile == "-" {
+			for _, n := range h.norms {
+				err := h.write(h.mode.outfile, n)
+				if err != nil {
+					return err
+				}
+			}
+		} else {
+			dir := h.mode.outfile
+			dir = strings.TrimSuffix(dir, ".ncd")
+			err := h.ctx.FileSystem().Mkdir(dir, 0o755)
```

Comment:
The filesystem permissions on this directory look too lax.

Wouldn't 0o644 suffice?